### ISSUE TITLE
Currently fairseq-py requires PyTorch version >= 0.4.0 ?

### ISSUE THREAD

**Issue by mali-nuist** [facebookresearch/fairseq-py#127]
I notice that README says: "**Currently fairseq-py requires PyTorch version >= 0.4.0**". But the latest pytorch version seems to be 0.3.1 ...

**Comment by mali-nuist** [comment]
and my startup command is :
CUDA_VISIBLE_DEVICES=0
python3  /DATA/sdb/pytorch/fairseq-py-dist/fairseq-py/train.py  ./prprocessed  --lr 0.25 --clip-norm 0.1 --dropout 0.2 --max-tokens 4000  --arch fconv --save-dir ./checkpoints --skip-invalid-size-inputs-valid-test

**Comment by travel-go** [comment]
Hello, Could you tell me how to deal with it?

**Comment by mali-nuist** [comment]
You should reinstall the pytorch from source code, because the lastest version[0.4 version] is not released yet.
-------------------------------------------------------
First, install pytorch from source code according this: 
    https://github.com/pytorch/pytorch#from-source.

Second, install nccl2, if you want to train your own fairseq model [not nccl1!!!!!!!!!!!]:
    http://docs.nvidia.com/deeplearning/sdk/nccl-install-guide/index.html#down

Finally, clone the latest fairseq-py and build the code:
    https://github.com/facebookresearch/fairseq-py  [the nccl page link in this page is nccl 1, do not install it]

**Comment by travel-go** [comment]
thank you for your reply!
